### PR TITLE
fix(addon-commerce): correct rounding for float values

### LINKED ISSUE #1930: 🐞 - Incorrect rounding amount
### Which @taiga-ui/* package(s) are the source of the bug?

addon-commerce

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/taiga-starter

### Is this issue blocking you?

Non-Blocking

### Description

Hi! I try show total amount in "amount" component. I use this value: 8918 + 10333.6 + 3527.78 + 805.62 + 140, but get wrong result (23824). It should be 23825

### Angular version

10.2.3

### Taiga UI version

2.32.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1930

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-commerce/components/money/money-options.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import {InjectionToken, ValueProvider} from '@angular/core';
 import {TuiCurrency} from '@taiga-ui/addon-commerce/enums';
 import {TuiCurrencyVariants, TuiMoneySignT} from '@taiga-ui/addon-commerce/types';
+import {CHAR_EN_DASH, CHAR_PLUS} from '@taiga-ui/cdk';
 import {TuiDecimalT} from '@taiga-ui/core';
 
+export type TuiSignSymbol = '' | typeof CHAR_EN_DASH | typeof CHAR_PLUS;
```
Comment (waterplea) [inline projects/addon-commerce/components/money/money-options.ts:7]: Looks like it should be a separate file

File: projects/addon-commerce/components/money/money.component.ts [hunk 2]
```diff
@@ -51,38 +47,8 @@ export class TuiMoneyComponent {
...
-            return '';
-        }
-
-        if (sign === 'force-negative' || (value < 0 && sign !== 'force-positive')) {
-            /** TODO(nsbarsukov): investigate if it should be replaced by {@link CHAR_HYPHEN} */
-            return CHAR_EN_DASH;
-        }
-
-        return CHAR_PLUS;
+    get signSymbol(): TuiSignSymbol {
```
Comment (waterplea) [inline projects/addon-commerce/components/money/money.component.ts:50]: Is it used somewhere?

File: projects/addon-commerce/components/money/money.template.html [hunk 3]
```diff
@@ -1,7 +1,16 @@
-<span automation-id="tui-money__sign">{{ signSymbol }}</span>
-<span automation-id="tui-money__integer-part">{{ integerPart }}</span>
+<span
+    automation-id="tui-money__sign"
+    [innerText]="value | tuiSignSymbol: sign"
```
Comment (waterplea) [inline projects/addon-commerce/components/money/money.template.html:3]: `textContent`? And below

File: projects/addon-commerce/components/money/pipes/fraction-part.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+export class TuiFractionPartPipe implements PipeTransform {
+    constructor(
+        @Inject(TUI_NUMBER_FORMAT) private readonly numberFormat: TuiNumberFormatSettings,
+    ) {}
+
+    transform(value: number, decimal: TuiDecimalT, precision: number): string {
+        const fraction = value.toFixed(precision).split('.')[1];
+
+        return decimal === 'never' ||
+            (parseInt(fraction, 10) === 0 && decimal !== 'always')
```
Comment (waterplea) [inline projects/addon-commerce/components/money/pipes/fraction-part.pipe.ts:14]: Let's move it to `const shouldShow = decimal === 'always' || !!parseInt(fraction, 10)`

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import {formatNumber, TUI_NUMBER_FORMAT, TuiNumberFormatSettings} from '@taiga-ui/core';
+
+@Pipe({name: 'tuiIntegerPart'})
+export class TuiIntegerPartPipe implements PipeTransform {
+    constructor(
+        @Inject(TUI_NUMBER_FORMAT) private readonly numberFormat: TuiNumberFormatSettings,
+    ) {}
+
+    transform(value: number, precision: number): string {
+        return formatNumber(
```
Comment (waterplea) [inline projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:11]: We should probably refactor this method with an `options` object :)

File: projects/addon-commerce/components/money/money.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-commerce/components/money/money.component.ts:50]: yes, in Hostlistener

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts (hunk 5 above)
Comment (splincode) [reply projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:11]: Sorry, could you suggest what I should do?

```ts
transform(value: number, { precision, decimalSeparator, thousandSeparator }): string {
```

or you mean

```ts
transform(value: number, precision: number): string {
    const {decimalSeparator, thousandSeparator} = this.numberFormat;

     return formatNumber(
            Math.floor(Math.abs(Number(value.toFixed(precision)))),
            null,
            decimalSeparator,
            thousandSeparator,
        );

```

File: projects/addon-commerce/components/money/pipes/fraction-part.pipe.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-commerce/components/money/pipes/fraction-part.pipe.ts]: fixed

File: projects/addon-commerce/components/money/money.template.html (hunk 3 above)
Comment (splincode) [reply projects/addon-commerce/components/money/money.template.html]: fixed

File: projects/addon-commerce/components/money/money-options.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-commerce/components/money/money-options.ts]: fixed

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts (hunk 5 above)
Comment (waterplea) [reply projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:11]: I mean `formatNumber` should be `formatNumber(value: number, option: Partial<AllTheStuff>)`. That's for the future, not for this PR.

**Review by waterplea — APPROVED** [review]

File: projects/addon-commerce/components/money/money.component.ts [hunk 6]
```diff
@@ -109,9 +79,9 @@ export class TuiMoneyComponent {
     }
 
     constructor(
-        @Inject(TUI_NUMBER_FORMAT)
-        private readonly numberFormat: TuiNumberFormatSettings,
         @Inject(TUI_MONEY_OPTIONS)
         private readonly options: TuiMoneyOptions,
+        @Inject(TuiSignSymbolPipe)
+        private readonly symbolTransformer: TuiSignSymbolPipe,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-commerce/components/money/money.component.ts]: fixed

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts (hunk 5 above)
Comment (splincode) [reply projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:11]: added todo

**Pull request merged**